### PR TITLE
B 02

### PR DESCRIPTION
**Pull request by MaksOnofriychuk**

### CODE REVIEW

File: src/redux/Table/index.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        .fill(0)
+        .map((_, i) => i + 1);
+
+      const rowsArray = Array(Number(action.payload.rows))
+        .fill(0)
+        .map((_, i) => i + 1);
+
+      const matrixArray = rowsArray.map((_, i) => {
+        return columnsArray.map((_, j) => ({
+          id: `${i} + ${j}`,
```
Comment (apiloian) [inline src/redux/Table/index.js]: are you sure ?

File: src/redux/Table/index.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  columnsArray: [],
+  formData: "",
+};
+
+export default function tableReducer(state = initialState, action) {
+  switch (action.type) {
+    case GET_TABLE_DATA: {
+      const columnsArray = Array(Number(action.payload.columns))
+        .fill(0)
+        .map((_, i) => i + 1);
```
Comment (apiloian) [inline src/redux/Table/index.js]: create a func for this operation.

File: src/redux/Table/index.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      };
+    }
+
+    case ADD_ROW:
+      return {
+        ...state,
+        matrixState: [
+          ...state.matrixState,
+          state.columnsArray.map((column, i) => ({
+            id: `${column}${i}`,
```
Comment (apiloian) [inline src/redux/Table/index.js]: don`t use variable " i " change to " index "

File: src/components/Table/columnAverage.jsx [hunk 4]
```diff
@@ -1,6 +1,8 @@
 import React from "react";
+import { useSelector } from "react-redux";
 
-const ColumnAverage = ({ matrixState }) => {
+const ColumnAverage = () => {
+  const { matrixState } = useSelector((state) => state.table);
   const sumByColumnsArray = matrixState.reduce((resultArr, arr) => {
     arr.forEach((cell, index) => {
```
Comment (apiloian) [inline src/components/Table/columnAverage.jsx:7]: 20я  строка  key

File: src/components/Table/Columns.jsx [hunk 5]
```diff
@@ -38,33 +39,19 @@ export const ButtonDelete = ({ matrixState, setMatrixState, index }) => {
...
 
-const Columns = ({ matrixState, setMatrixState }) => {
+const Columns = () => {
+  const { matrixState } = useSelector((state) => state.table);
   return (
     <>
       {matrixState.map((rows, index) => {
         return (
-          <tr key={rows[0].value}>
+          <tr key={`${rows[0].value} ${index}`}>
```
Comment (apiloian) [inline src/components/Table/Columns.jsx]: пробел

File: src/redux/Table/index.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+      return {
+        ...state,
+        matrixState: newMatrixState,
+      };
+    }
+
+    case DELETE_ROW: {
+      const newMatrixState = state.matrixState.filter(
+        (_, i) => i !== action.payload
```
Comment (apiloian) [inline src/redux/Table/index.js]: don`t delete by index  only by id

**Review by arturiop — APPROVED** [review]

**Pull request merged**